doc/classes/String.xml

```diff
@@ -714,65 +804,72 @@
 			<param index="1" name="allow_empty" type="bool" default="true" />
 			<param index="2" name="maxsplit" type="int" default="0" />
 			<description>
-				Splits the string by a [param delimiter] string and returns an array of the substrings. The [param delimiter] can be of any length. If [param delimiter] is an empty string, each substring will be a single character.
-				If [param allow_empty] is [code]true[/code], and there are two adjacent delimiters in the string, it will add an empty string to the array of substrings at this position.
-				If [param maxsplit] is specified, it defines the number of splits to do from the left up to [param maxsplit]. The default value of [code]0[/code] means that all items are split.
-				If you need only one element from the array at a specific index, [method get_slice] is a more performant option.
+				Splits the string using a [param delimiter] and returns an array of the substrings. If [param delimiter] is an empty string, each substring will be a single character. This method is the opposite of [method join].
+				If [param allow_empty] is [code]false[/code], empty strings between adjacent delimiters are excluded from the array.
+				If [param maxsplit] is greater than [code]0[/code], the number of splits may not exceed [param maxsplit]. By default, the entire string is split.
 				[b]Example:[/b]
 				[codeblocks]
 				[gdscript]
-				var some_string = "One,Two,Three,Four"
-				var some_array = some_string.split(",", true, 1)
-				print(some_array.size()) # Prints 2
-				print(some_array[0]) # Prints "Four"
-				print(some_array[1]) # Prints "Three,Two,One"
+				var some_array = "One,Two,Three,Four".split(",", true, 2)
+
+				print(some_array.size()) # Prints 3
+				print(some_array[0])     # Prints "One"
+				print(some_array[1])     # Prints "Two"
+				print(some_array[2])     # Prints "Three,Four"
 				[/gdscript]
 				[csharp]
-				var someString = "One,Two,Three,Four";
-				var someArray = someString.Split(",", true); // This is as close as it gets to Godots API.
-				GD.Print(someArray[0]); // Prints "Four"
-				GD.Print(someArray[1]); // Prints "Three,Two,One"
+				// C#'s native Split() does not support the `maxsplit` parameter.
```

Comment:
Not sure if it's relevant to specify "native" here, it may confuse users. BTW, there is a `Split` method in `System.String` but we also implement our own in `StringExtensions`, neither support `maxsplit`.

Reply:
It is worth not specifying then, yeah.